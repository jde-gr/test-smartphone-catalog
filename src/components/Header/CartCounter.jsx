import React from 'react';
import { useSelector } from 'react-redux';
import { SECONDARY } from '../../core/constants';
import { Badge } from './styles/CardCounter.styles';

const CartCounter = () => {
  const itemCount = useSelector((state) => state.cart.itemCount);
  return (
    <div>
      <Badge variantColor={SECONDARY} data-testid='cartItemQty'>
        {itemCount}
      </Badge>
    </div>
  );
};

export default CartCounter;
