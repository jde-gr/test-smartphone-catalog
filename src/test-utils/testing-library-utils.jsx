import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store';

const queryClient = new QueryClient();

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          {children}
        </QueryClientProvider>
      </Provider>
    </BrowserRouter>
  );
};
const renderWithContext = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { renderWithContext as render };
