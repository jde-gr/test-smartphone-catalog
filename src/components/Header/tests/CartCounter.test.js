import { render } from '../../../test-utils/testing-library-utils';
import CartCounter from '../CartCounter';

test('renders correctly', () => {
  const { container } = render(<CartCounter />);
  expect(container).toMatchSnapshot();
});
