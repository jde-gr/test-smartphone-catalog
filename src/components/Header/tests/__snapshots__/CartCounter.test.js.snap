// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <span
      class="css-cdjgyj"
      data-testid="cartItemQty"
    >
      0
    </span>
  </div>
</div>
`;
